### DISCUSSION TITLE
[Turborepo] How can I bypass .gitignore? The cache still hits even when the file has been modified.

### DISCUSSION THREAD

**Discussion by Cow258** [vercel/turborepo#3879] [Help] [answered]
I have two JSON files `site.config.json` and `i18n.locale.json`, 
that are built from a database and then incorporated into a Next.js app for SSR support. 
Because the content of these files is dynamic, 
I've added them to `.gitignore` to prevent them from affecting each commit. 
However, Turbo Repo won't hash these files when they're included in `.gitignore`, 
which means that even if the files are modified, 
the next build will still hit the cache.

### Based on my testing, the following are my results:
1. When `site.config.json` and `i18n.locale.json` are included in `.gitignore`
   - Cache hits occur even when these files are modified.
2. When `site.config.json` and `i18n.locale.json` are not included in `.gitignore`
   - Cache bypass occurs when these files are modified, but cache hits occur when these files are not modified.

### Currently status
I have added `site.config.json` and `i18n.locale.json` to `.gitignore` to enable the cache mechanism to work, but this is affecting every commit and the repo is never clean. I could turn off the cache with "build": {"cache": false}, but is there a solution for fixing this issue?

Please let me know if there are any mistakes in my configuration.

### File Path
`site.config.json` - `./apps/web/site.config.json`
`i18n.locale.json` - `./apps/web/i18n.locale.json`

`pnpm-workspace.yaml`
```yml
packages:
  - "apps/*"
  - "packages/*"
```

`turbo.json`
```json
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "inputs": [
        "i18n.locales.json",
        "i18n.locales.d.ts",
        "site.config.json",
        "**/*.js",
        "**/*.jsx",
        "**/*.json",
        "**/*.svg"
      ],
      "outputs": [
        ".next/**",
        "!.next/cache/**"
      ]
    },
  }
}
```

**Comment by Cow258** [comment]
I'm still stuck on this issue?
Any suggestion?
Thanks a lot! 🙏🏻

**Comment by Cow258** [comment]
I'm still stuck on this issue?
Any suggestion?
Thanks a lot! 🙏🏻

**Comment by yoDon** [comment]
You might be able to leverage another bug in the caching system. According to https://github.com/vercel/turbo/issues/1466 if you list files in a nested .gitignore file (eg. ./apps/web/.gitignore), the turbo repo caching system fails to identify the files in the nested .gitignore as being ignored.

**Reply by Cow258** [reply]
You mean moving the nested .gitignore content to root will bypassed this issue? 👀 🤔
I will try it next week.
thanks a lot

**Reply by Cow258** [reply]
Thanks, It work.

**Comment by mehulkar** [comment, marked as answer]
I just wrote a test that seems to prove that you can use gitignored files as inputs and it will be part of the cache. It's possible I got the test wrong though, could you help review? https://github.com/vercel/turbo/pull/6669
